contributors/design-proposals/resource-management/device-plugin.md

```diff
@@ -340,37 +325,39 @@ message Device {
 We want Kubelet as well as the Device Plugins to recover from failures
 that may happen on any side of this protocol.
 
-At the communication level, gRPC is a very strong piece of software and
-is able to ensure that if failure happens it will try it's best to recover
-through exponential backoff reconnection and Keep Alive checks.
-
 The proposed mechanism intends to replace any device specific handling in
-Kubelet. Therefore in general, device plugin failure or upgrade means that
-Kubelet is not able to accept any pod requesting a Device until the upgrade
-or failure finishes.
+Kubelet. Therefore in general, device plugin failure means that
+Kubelet is not able to accept any new pod requesting a Device until the
+failure finishes.
 
 If a device fails, the Device Plugin should signal that through the
-`ListAndWatch` gRPC stream. We then expect Kubelet to fail the Pod.
+`ListAndWatch` gRPC stream. We then expect Kubelet to avoid scheduling pods on the
+failed device.
```

Comment:
What happens if the Kubelet cannot find another healthy device to satisfy the container resource request?